Fix problem with incorrect time definition

TouDead opened the pull request:
I try to use 1h30m and get 130 hours. So I write this code to fix this

TouDead committed:
Fix problem with incorrect time definition

I try to use 1h30m and get 130 hours. So I write this code to fix this

KarlOfDuty commented:
This is a great addition, thanks!

However, the code doesn't seem to compile at the moment, so you will need to fix that first.

KarlOfDuty commented on SCPDiscordPlugin/Utilities.cs:
```diff
@@ -106,62 +106,57 @@ public static LinkedList<string> ParseListIntoMessages(List<string> listItems)
+                switch (unit)
+                {
+                    case 's':
+                        humanReadableDuration = amount + " second";
```
This will not return the correct time to the command user anymore, but I think you could just skip using it in the for loop and use the SecondsToCompoundTime() utilities function at the end of the loop to get the human readable version.

KarlOfDuty commented on SCPDiscordPlugin/Utilities.cs:
```diff
@@ -106,62 +106,57 @@ public static LinkedList<string> ParseListIntoMessages(List<string> listItems)
-		}
+        {
+            TimeSpan timeSpanDuration = TimeSpan.Zero;
+            var matches = Regex.Matches(duration, @"(\d+)([smhdwMy])");
```
As I mentioned before, you forgot to include the regex functions so it will not compile atm.

KarlOfDuty replied:
And unless the type name is very long I would like the variables to use explicit types.

TouDead committed:
Total rework

I thought a little, took into account the wishes and found a more elegant solution to the problem

TouDead committed:
update using

I forgot to update using again

KarlOfDuty reviewed (changes requested):
Actually I think your first solution was better. While I think your second version may be an improvement in performance, that is not a priority in this case.

Your first version was very readable and easy to understand for people who are not very familiar with the project so I would rather use that one plus the fixes we discussed.

What do you think about this version? https://github.com/KarlOfDuty/SCPDiscord/commit/b11a8116f46d446577bd59e21358ad53ec32d57f

If you want to use this version but get some merge conflicts I can solve them for you if you wish.

KarlOfDuty commented:
But if you think your second version is better you can make a version of it that uses the SecondsToCompoundTime function instead of implementing that code yourself there, I'd be fine with that too.

TouDead committed:
Changing the way you calculate time

TouDead commented:
Was busy for a few days but came back to finish the job. I took into account the wishes for the readability of the code and brought it into working condition, with all the necessary corrections on my part and yours.

KarlOfDuty committed:
Moved humanReadableDuration and added error handlng

KarlOfDuty reviewed (approved):
Very nice, looks good!

I made a small change where invalid strings give an error instead of a 0 second duration, and moved the human readable duration out of this function. With your improvement I don't think that parameter is needed anymore.

Thank you very much for the contribution!

KarlOfDuty commented:
The dev build for this PR is available here: https://jenkins.karlofduty.com/blue/organizations/jenkins/NWAPI%2FSCPDiscord/detail/master/213/artifacts
